multimap/src/test/java/org/infinispan/multimap/impl/DistributedMultimapCacheTest.java

```diff
@@ -0,0 +1,382 @@
+package org.infinispan.multimap.impl;
+
+import static java.lang.String.format;
+import static org.infinispan.functional.FunctionalTestUtils.await;
+import static org.infinispan.multimap.impl.MultimapTestUtils.JULIEN;
+import static org.infinispan.multimap.impl.MultimapTestUtils.NAMES_KEY;
+import static org.infinispan.multimap.impl.MultimapTestUtils.OIHANA;
+import static org.infinispan.multimap.impl.MultimapTestUtils.RAMON;
+import static org.infinispan.multimap.impl.MultimapTestUtils.assertMultimapCacheSize;
+import static org.infinispan.multimap.impl.MultimapTestUtils.putValuesOnMultimapCache;
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.assertFalse;
+import static org.testng.AssertJUnit.assertNotNull;
+import static org.testng.AssertJUnit.assertTrue;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.infinispan.Cache;
+import org.infinispan.commons.dataconversion.Encoder;
+import org.infinispan.commons.dataconversion.EncodingUtils;
+import org.infinispan.commons.dataconversion.Wrapper;
+import org.infinispan.configuration.cache.CacheMode;
+import org.infinispan.container.DataContainer;
+import org.infinispan.container.entries.CacheEntry;
+import org.infinispan.container.entries.ImmortalCacheEntry;
+import org.infinispan.container.entries.InternalCacheEntry;
+import org.infinispan.distribution.BaseDistFunctionalTest;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.multimap.api.EmbeddedMultimapCacheManagerFactory;
+import org.infinispan.multimap.api.MultimapCache;
+import org.infinispan.multimap.api.MultimapCacheManager;
+import org.infinispan.remoting.transport.Address;
+import org.testng.annotations.Test;
+
+@Test(groups = "functional", testName = "distribution.DistributedMultimapCacheTest")
+public class DistributedMultimapCacheTest extends BaseDistFunctionalTest<String, Collection<User>> {
+
+   protected Map<Address, MultimapCache<String, User>> multimapCacheCluster = new HashMap<>();
+
+   protected boolean fromOwner;
+
+   public DistributedMultimapCacheTest fromOwner(boolean fromOwner) {
+      this.fromOwner = fromOwner;
+      return this;
+   }
+
+   @Override
+   protected String[] parameterNames() {
+      return concat(super.parameterNames(), "fromOwner");
+   }
+
+   @Override
+   protected Object[] parameterValues() {
+      return concat(super.parameterValues(), fromOwner ? Boolean.TRUE : Boolean.FALSE);
+   }
+
+   @Override
+   public Object[] factory() {
+      List testsToRun = new ArrayList<>();
```

Comment:
no, this is because I went to tx tests and I was about to do something similar to create tests; I can refactor that, doesn't bother me